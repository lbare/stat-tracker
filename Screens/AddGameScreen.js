import { View, Text, TouchableOpacity, TextInput, Alert } from "react-native";
import { useState } from "react";
import DateTimePicker from "@react-native-community/datetimepicker";

const AddGameScreen = ({ navigation }) => {
  const [opponent, setOpponent] = useState("");
  const [date, setDate] = useState(new Date());
  const [time, setTime] = useState(new Date());
  const [home, setHome] = useState(null);

  return (
    <View className="flex-1 justify-center items-center">
      <TextInput
        className="bg-gray-100 border border-gray-500 mx-10 p-4 rounded-xl w-1/2 text-center"
        title="Opponent"
        placeholder="Opponent"
        blurOnSubmit={false}
        secureTextEntry={false}
        keyboardType="default"
        textContentType="none"
        returnKeyType="next"
        onChangeText={(value) => setOpponent(value)}
      />
      <View className="justify-center items-center h-12 w-1/2 border border-grey-500 rounded-xl">
        <DateTimePicker
          value={date}
          mode="date"
          display="default"
          onChange={(event, selectedDate) => setDate(selectedDate)}
          style={{}}
        />
      </View>
      <View className="justify-center items-center h-12 w-1/2 border border-grey-500 rounded-xl">
        <DateTimePicker
          value={time}
          mode="time"
          display="default"
          onChange={(event, selectedDate) => setTime(selectedDate)}
          style={{}}
        />
      </View>
      <View className="flex-row justify-evenly p-6 w-full">
        <TouchableOpacity
          className={`bg-gray-100 border border-gray-500 rounded-xl w-1/3 h-14 justify-center ${
            home ? "bg-blue-500 border-blue-700" : ""
          }`}
          onPress={() => setHome(true)}
        >
          <Text className={`text-center text-xl ${home ? "text-white" : ""}`}>
            Home
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          className={`bg-gray-100 border border-gray-500 rounded-xl w-1/3 h-14 justify-center ${
            home === false ? "bg-blue-500 border-blue-700" : ""
          }`}
          onPress={() => setHome(false)}
        >
          <Text
            className={`text-center text-xl ${
              home === false ? "text-white" : ""
            }`}
          >
            Away
          </Text>
        </TouchableOpacity>
      </View>

      <TouchableOpacity
        className="border rounded-xl w-1/3 h-14 justify-center bg-blue-500 border-blue-700"
        onPress={() => {
          Alert.alert("Confirm", "Are you sure you want to add this game?", [
            {
              text: "Cancel",
              style: "cancel",
            },
            {
              text: "Add",
              onPress: () => {
                const datetime = new Date(
                  date.getFullYear(),
                  date.getMonth(),
                  date.getDate(),
                  time.getHours(),
                  time.getMinutes()
                ).toISOString();
                console.log(opponent, datetime, home);
              },
            },
          ]);
        }}
      >
        <Text className="text-center text-xl text-white">Add Game</Text>
      </TouchableOpacity>
    </View>
  );
};

export default AddGameScreen;
