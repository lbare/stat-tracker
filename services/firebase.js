import { getApps, initializeApp, getApp } from "firebase/app";
import {
  getAuth,
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  onAuthStateChanged,
} from "firebase/auth";
import {
  getFirestore,
  collection,
  addDoc,
  docRef,
  doc,
  getDoc,
  getDocs,
  setDoc,
} from "firebase/firestore";
import uuid from "react-native-uuid";

const firebaseConfig = {
  apiKey: "AIzaSyBOgTaYS8HJA1uFEebcKBFukJ72Je_j5GM",
  authDomain: "stat-tracker-9f25a.firebaseapp.com",
  projectId: "stat-tracker-9f25a",
  storageBucket: "stat-tracker-9f25a.appspot.com",
  messagingSenderId: "491322406723",
  appId: "1:491322406723:web:c3c263d8e4907f92a621f6",
};

const app = !getApps().length ? initializeApp(firebaseConfig) : getApp();
const auth = getAuth(app);
const db = getFirestore(app);

/* AUTH */
export const login = async (email, password) => {
  try {
    const userCredentials = await signInWithEmailAndPassword(
      auth,
      email,
      password
    );
    const uid = userCredentials.user.uid;
    const docSnap = await getDoc(doc(db, "users", uid));
    if (docSnap.exists()) {
      return docSnap.data();
    } else {
      console.log("No such document");
      return null;
    }
  } catch (error) {
    alert(error);
    return null;
  }
};

export const addUser = async (email, password) => {
  try {
    const response = await createUserWithEmailAndPassword(
      auth,
      email,
      password
    );
    const uid = response.user.uid;
    const data = { uid: uid, email };
    const result = await setDoc(doc(db, "users", uid), data);
    return result;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

export const addSeason = async (season, id = null) => {
  try {
    const docRef = id ? doc(db, "seasons", id) : doc(collection(db, "seasons"));
    const result = await setDoc(docRef, season);
    console.log("Season added successfully!");
    return result;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

export const addGame = async (game, id = null) => {
  try {
    const docRef = id ? doc(db, "games", id) : doc(collection(db, "games"));
    const result = await setDoc(docRef, game);
    console.log("Game added successfully!");
    return result;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

export const addAtBat = async (atBat, id = null) => {
  try {
    const docRef = id ? doc(db, "atBats", id) : doc(collection(db, "atBats"));
    const result = await setDoc(docRef, atBat);
    console.log("At-bat added successfully!");
    return result;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

/* WRITES */
export const getAllSeasons = async () => {
  try {
    const snapshot = await getDocs(collection(db, "seasons"));
    const seasons = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
    return seasons;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

export const getAllGames = async () => {
  try {
    const snapshot = await getDocs(collection(db, "games"));
    const games = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
    return games;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

export const getAllAtBats = async () => {
  try {
    const snapshot = await getDocs(collection(db, "atBats"));
    const atBats = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
    return atBats;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

export const getAllGamesBySeason = async (seasonId) => {
  try {
    const snapshot = await getDocs(
      query(collection(db, "games"), where("seasonId", "==", seasonId))
    );
    const games = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
    return games;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

export const getAllAtBatsBySeason = async (seasonId) => {
  try {
    const snapshot = await getDocs(
      query(collection(db, "atBats"), where("seasonId", "==", seasonId))
    );
    const atBats = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
    return atBats;
  } catch (error) {
    console.error(error);
    throw error;
  }
};

export const getAllAtBatsByGame = async (gameId) => {
  let atBats = [];

  const querySnapshot = await collection(db, "atBats")
    .where("gameId", "==", gameId)
    .get();

  querySnapshot.forEach((doc) => {
    atBats.push(doc.data());
  });

  return atBats;
};

export const generateId = () => {
  return uuid.v4();
};

export {
  db,
  auth,
  collection,
  addDoc,
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  docRef,
  doc,
  getDoc,
  setDoc,
  onAuthStateChanged,
};
