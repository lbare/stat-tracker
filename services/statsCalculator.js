export class StatsCalculator {
  constructor(atBats, pitching) {
    this.atBats = atBats || [];
    this.pitching = pitching || [];

    // batting
    this.H = 0;
    this.Do = 0;
    this.Tr = 0;
    this.HR = 0;
    this.XBH = 0;
    this.TB = 0;
    this.PA = 0;
    this.AB = 0;
    this.Out = 0;
    this.R = 0;
    this.RBI = 0;
    this.BB = 0;
    this.IBB = 0;
    this.HBP = 0;
    this.K = 0;
    this.KL = 0;
    this.KS = 0;
    this.GIDP = 0;
    this.SB = 0;
    this.CS = 0;
    this.SF = 0;
    this.E = 0;
    this.SAC = 0;

    this.AVG = 0.0;
    this.OBP = 0.0;
    this.SLG = 0.0;
    this.OPS = 0.0;
    this.BABIP = 0.0;
    this.ISO = 0.0;
    this.BBpK = 0.0;
    this.BBper = 0.0;
    this.Kper = 0.0;

    // pitching
    this.IP = 0;
    this.BF = 0;
    this.G = 0;
    this.GS = 0;
    this.CG = 0;
    this.QS = 0;
    this.ER = 0;
    this.RA = 0;
    this.BBA = 0;
    this.IBBA = 0;
    this.HBPA = 0;
    this.PK = 0;
    this.EA = 0;
    this.KA = 0;
    this.HA = 0;
    this.W = 0;
    this.L = 0;

    this.ERA = 0.0;
    this.WHIP = 0.0;
    this.BAA = 0.0;
    this.K7 = 0.0;
    this.BB7 = 0.0;
    this.H7 = 0.0;
    this.KpBB = 0.0;
    this.BABIPA = 0.0;

    // fielding
    this.PO = 0;
    this.A = 0;
    this.E = 0;

    this.processData();
  }

  processData() {
    this.calculateBattingTotals();
    this.pitching.length === 0
      ? this.calculateSinglePitchingTotals()
      : this.calculateMultiplePitchingTotals();
    this.calculateAggregateTotals();
  }

  calculateAggregateTotals() {
    this.AVG = this.getAVG();
    this.OBP = this.getOBP();
    this.SLG = this.getSLG();
    this.OPS = this.getOPS();
    this.BABIP = this.getBABIP();
    this.ISO = this.getISO();
    this.BBpK = this.getBBperK();
    this.BBper = this.getBBpercent();
    this.Kper = this.getKpercent();

    this.ERA = this.getERA();
    this.WHIP = this.getWHIP();
    this.BAA = this.getBAA();
    this.K7 = this.getK7();
    this.BB7 = this.getBB7();
    this.H7 = this.getH7();
    this.KpBB = this.getKperBB();
    this.BABIPA = this.getBABIPA();
  }

  calculateBattingTotals() {
    const resultMap = {
      "1B": { H: 1, Si: 1, AB: 1, TB: 1 },
      "2B": { H: 1, Do: 1, AB: 1, TB: 2, XBH: 1 },
      "3B": { H: 1, Tr: 1, AB: 1, TB: 3, XBH: 1 },
      HR: { H: 1, HR: 1, AB: 1, TB: 4, XBH: 1 },
      BB: { BB: 1 },
      IBB: { IBB: 1 },
      Out: { AB: 1 },
      SF: { PA: 1 },
      SAC: { PA: 1 },
      HBP: { HBP: 1 },
      KS: { K: 1, KS: 1, AB: 1 },
      KL: { K: 1, KL: 1, AB: 1 },
      GIDP: { GIDP: 1, AB: 1 },
      E: { E: 1, AB: 1 },
    };

    this.atBats.forEach((atBat) => {
      this.PA++;

      if (atBat.result in resultMap) {
        const updates = resultMap[atBat.result];
        for (const field in updates) {
          this[field] += updates[field];
        }
      }

      this.R += atBat.runScored ? 1 : 0;
      this.RBI += atBat.RBI;
      this.SB += atBat.stolenBases;
      this.CS += atBat.caughtStealing;
    });
  }

  calculateSinglePitchingTotals() {
    this.IP = this.pitching.inningsPitched;
    this.BP =
      this.pitching.inningsPitched * 3 +
      this.pitching.walks +
      this.pitching.intentionalWalks +
      this.pitching.hitByPitch +
      this.pitching.hits +
      this.pitching.errorsAgainst;
    this.G = this.pitching.inningsPitched > 0 ? 1 : 0;
    this.GS = this.pitching.started ? 1 : 0;
    this.CG = this.pitching.completeGame ? 1 : 0;
    this.QS =
      this.pitching.inningsPitched > 4 && this.pitching.earnedRuns < 3 ? 1 : 0;
    this.ER = this.pitching.earnedRuns;
    this.RA = this.pitching.runsAllowed;
    this.BBA = this.pitching.walks;
    this.IBBA = this.pitching.intentionalWalks;
    this.HBPA = this.pitching.hitByPitch;
    this.PK = this.pitching.pickoffs;
    this.EA = this.pitching.errorsAgainst;
    this.KA = this.pitching.strikeouts;
    this.HA = this.pitching.hits;
    this.W = this.pitching.win ? 1 : 0;
    this.L = this.pitching.lose ? 1 : 0;
  }

  calculateMultiplePitchingTotals() {
    this.pitching.forEach((appearance) => {
      this.IP += appearance.inningsPitched;
      this.BP +=
        appearance.inningsPitched * 3 +
        appearance.walks +
        appearance.intentionalWalks +
        appearance.hitByPitch +
        appearance.hits +
        appearance.errorsAgainst;
      this.G += appearance.inningsPitched > 0 ? 1 : 0;
      this.GS += appearance.started ? 1 : 0;
      this.CG += appearance.completeGame ? 1 : 0;
      this.QS +=
        appearance.inningsPitched > 4 && appearance.earnedRuns < 3 ? 1 : 0;
      this.ER += appearance.earnedRuns;
      this.RA += appearance.runsAllowed;
      this.BBA += appearance.walks;
      this.IBBA += appearance.intentionalWalks;
      this.HBPA += appearance.hitByPitch;
      this.PK += appearance.pickoffs;
      this.EA += appearance.errorsAgainst;
      this.KA += appearance.strikeouts;
      this.HA += appearance.hits;
      this.W += appearance.win ? 1 : 0;
      this.L += appearance.lose ? 1 : 0;
    });
  }

  /************/
  /* Hitting */
  /************/

  getH() {
    return this.H || 0;
  }

  get1B() {
    return this.Si || 0;
  }

  get2B() {
    return this.Do || 0;
  }

  get3B() {
    return this.Tr || 0;
  }

  getHR() {
    return this.HR || 0;
  }

  getXBH() {
    return this.XBH || 0;
  }

  getTB() {
    return this.TB || 0;
  }

  getPA() {
    return this.PA || 0;
  }

  getAB() {
    return this.AB || 0;
  }

  getR() {
    return this.R || 0;
  }

  getRBI() {
    return this.RBI || 0;
  }

  getBB() {
    return this.BB || 0;
  }

  getIBB() {
    return this.IBB || 0;
  }

  getHBP() {
    return this.HBP || 0;
  }

  getK() {
    return this.K || 0;
  }

  getGIDP() {
    return this.GIDP || 0;
  }

  getSB() {
    return this.SB || 0;
  }

  getCS() {
    return this.CS || 0;
  }

  getSF() {
    return this.SF || 0;
  }

  getE() {
    return this.E || 0;
  }

  getSAC() {
    return this.SAC || 0;
  }

  getAVG() {
    return this.AB === 0 ? 0 : this.H / this.AB;
  }

  getOBP() {
    return this.PA === 0
      ? 0
      : (this.H + this.BB + this.HBP) /
          (this.AB + this.BB + this.HBP + this.SF);
  }

  getSLG() {
    return this.AB === 0 ? 0 : this.TB / this.AB;
  }

  getOPS() {
    return this.OBP + this.SLG || 0;
  }

  getBABIP() {
    return this.AB === 0
      ? 0
      : (this.H - this.HR) / (this.AB - this.K - this.HR);
  }

  getISO() {
    return this.AB === 0 ? 0 : this.SLG - this.AVG;
  }

  getBBperK() {
    return this.K === 0 ? 0 : this.BB / this.K;
  }

  getBBpercent() {
    return this.PA === 0 ? 0 : this.BB / this.PA;
  }

  getKpercent() {
    return this.PA === 0 ? 0 : this.K / this.PA;
  }

  /************/
  /* Pitching */
  /************/

  getIP() {
    return this.IP;
  }

  getBF() {
    return this.BF;
  }

  getG() {
    return this.G;
  }

  getGS() {
    return this.GS;
  }

  getCG() {
    return this.CG;
  }

  getQS() {
    return this.QS;
  }

  getER() {
    return this.ER;
  }

  getRA() {
    return this.RA;
  }

  getBBA() {
    return this.BBA;
  }

  getIBBA() {
    return this.IBBA;
  }

  getHBPA() {
    return this.HBPA;
  }

  getPK() {
    return this.PK;
  }

  getEA() {
    return this.EA;
  }

  getKA() {
    return this.KA;
  }

  getHA() {
    return this.HA;
  }

  getW() {
    return this.W;
  }

  getL() {
    return this.L;
  }

  getERA() {
    return this.IP === 0 ? 0 : (this.ER * 9) / this.IP;
  }

  getWHIP() {
    return this.IP === 0 ? 0 : (this.BBA + this.HA) / this.IP;
  }

  getBAA() {
    return this.IP === 0 ? 0 : (this.HA * 7) / this.IP;
  }

  getK7() {
    return this.IP === 0 ? 0 : (this.KA * 7) / this.IP;
  }

  getBB7() {
    return this.IP === 0 ? 0 : (this.BBA * 7) / this.IP;
  }

  getH7() {
    return this.IP === 0 ? 0 : (this.HA * 7) / this.IP;
  }

  getKperBB() {
    return this.BBA === 0 ? 0 : this.KA / this.BBA;
  }

  getBABIPA() {
    return this.IP === 0 ? 0 : (this.HA - this.HRA) / (this.IP - this.KA);
  }
}
