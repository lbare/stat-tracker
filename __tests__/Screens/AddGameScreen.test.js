import React from "react";
import renderer from "react-test-renderer";
import AddGameScreen from "../../Screens/AddGameScreen";
import { UserContext } from "../../services/UserContext";

const mockUserContext = {
  user: [],
  setUser: jest.fn(),
  userSeasons: [],
  setUserSeasons: jest.fn(),
  userGames: [],
  setUserGames: jest.fn(),
  userAtBats: 0,
  setUserAtBats: jest.fn(),
};

jest.mock("react-native-step-indicator");

describe("<AddGameScreen />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <UserContext.Provider value={mockUserContext}>
          <AddGameScreen />
        </UserContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
