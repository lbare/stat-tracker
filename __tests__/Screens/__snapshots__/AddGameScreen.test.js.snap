// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddGameScreen /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "width": "100%",
          },
          {
            "alignItems": "center",
          },
          {
            "paddingBottom": 64,
            "paddingTop": 64,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#6b7280",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 1,
            "borderLeftColor": "#6b7280",
            "borderLeftWidth": 1,
            "borderRightColor": "#6b7280",
            "borderRightWidth": 1,
            "borderTopColor": "#6b7280",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "opacity": 1,
            "width": "50%",
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "center",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
            ]
          }
        >
          Brewers
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "height": 48,
          },
          {
            "width": "50%",
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
          },
        ]
      }
    >
      <RNDateTimePicker
        date={1680656139133}
        displayIOS="default"
        enabled={true}
        mode="date"
        onChange={[Function]}
        onPickerDismiss={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={{}}
      />
    </View>
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "height": 48,
          },
          {
            "width": "50%",
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
          },
        ]
      }
    >
      <RNDateTimePicker
        date={1680656139133}
        displayIOS="default"
        enabled={true}
        mode="time"
        onChange={[Function]}
        onPickerDismiss={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={{}}
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "space-evenly",
          },
          {
            "paddingBottom": 24,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 24,
          },
          {
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#f3f4f6",
            "borderBottomColor": "#6b7280",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 1,
            "borderLeftColor": "#6b7280",
            "borderLeftWidth": 1,
            "borderRightColor": "#6b7280",
            "borderRightWidth": 1,
            "borderTopColor": "#6b7280",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 1,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "width": "33.333333%",
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "center",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
            ]
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#f3f4f6",
            "borderBottomColor": "#6b7280",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 1,
            "borderLeftColor": "#6b7280",
            "borderLeftWidth": 1,
            "borderRightColor": "#6b7280",
            "borderRightWidth": 1,
            "borderTopColor": "#6b7280",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 1,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "width": "33.333333%",
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "center",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
            ]
          }
        >
          Away
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#3b82f6",
          "borderBottomColor": "#1d4ed8",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderBottomWidth": 1,
          "borderLeftColor": "#1d4ed8",
          "borderLeftWidth": 1,
          "borderRightColor": "#1d4ed8",
          "borderRightWidth": 1,
          "borderTopColor": "#1d4ed8",
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "borderTopWidth": 1,
          "height": 56,
          "justifyContent": "center",
          "opacity": 1,
          "width": "33.333333%",
        }
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "center",
            },
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "color": "#fff",
            },
          ]
        }
      >
        Add Game
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
